pylint/plugins/hass_enforce_type_hints.py

```diff
@@ -8,10 +8,15 @@
 from pylint.checkers import BaseChecker
 from pylint.lint import PyLinter
 
+from homeassistant.backports.enum import StrEnum
 from homeassistant.const import Platform
 
-DEVICE_CLASS = object()
-UNDEFINED = object()
+
+class SentinelValues(StrEnum):
+    """Sentinel values"""
+
+    UNDEFINED = "***UNDEFINED"
```

Comment:
Tbh I'm not really sure anymore my suggestion to use `StrEnum` was helpful here. Do you think it's an improvement? I'd be fine with reverting it if you want.

Reply:
I think it's ok to keep. There might be other values we need to handle in a special case.